<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="VLz" failures="1" tests="13" name="com.juaracoding.tajuaracoding.laporan.KehadiranTest" time="204.903" errors="1" timestamp="2025-08-27T22:08:52 WIB" skipped="0">
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testExportFunctionality" time="10.020">
    <failure message="Start Date tidak sesuai. Expected: 20-08-2025, Actual: 20 Agt 2025 expected [true] but found [false]" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Start Date tidak sesuai. Expected: 20-08-2025, Actual: 20 Agt 2025 expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertTrue(Assert.java:57)
at com.juaracoding.tajuaracoding.pages.actions.DateFilterAction.execute(DateFilterAction.java:57)
at com.juaracoding.tajuaracoding.pages.actions.CompositeActionBuilder.execute(CompositeActionBuilder.java:110)
at com.juaracoding.tajuaracoding.laporan.KehadiranTest.testExportFunctionality(KehadiranTest.java:132)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:308)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:71)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:113)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
]]>
    </failure>
  </testcase> <!-- testExportFunctionality -->
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testKehadiranPageVerification" time="8.888"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testDateRangeFilter" time="15.535"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testPositiveSearch" time="11.887"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testNegativeSearch" time="13.672"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testPositiveSearchWithDateRangeFilter" time="17.171"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testPositiveUnitFilter" time="15.721"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testNegativeUnitFilter" time="12.759">
    <error message="Action failed: Unit filter gagal untuk: UnitTidakAda | Error: Hasil filter ada padahal seharusnya kosong untuk: UnitTidakAda" type="java.lang.RuntimeException">
      <![CDATA[java.lang.RuntimeException: Action failed: Unit filter gagal untuk: UnitTidakAda | Error: Hasil filter ada padahal seharusnya kosong untuk: UnitTidakAda
at com.juaracoding.tajuaracoding.pages.actions.CompositeActionBuilder.execute(CompositeActionBuilder.java:112)
at com.juaracoding.tajuaracoding.laporan.KehadiranTest.testNegativeUnitFilter(KehadiranTest.java:74)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:308)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:71)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:113)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
]]>
    </error>
  </testcase> <!-- testNegativeUnitFilter -->
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testPositiveUnitFilterWithDateRangeFilter" time="21.058"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testExportFunctionalityCancel" time="15.678"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testCombinationFilter" time="26.068"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testResetAllFilters" time="20.934"/>
  <testcase classname="com.juaracoding.tajuaracoding.laporan.KehadiranTest" name="testStatusCheck" time="15.512"/>
</testsuite> <!-- com.juaracoding.tajuaracoding.laporan.KehadiranTest -->
